src/main/java/org/openrewrite/java/dependencies/RelocatedDependencyCheck.java

```diff
@@ -130,181 +128,177 @@ public TreeVisitor<?, ExecutionContext> getVisitor(Accumulator acc) {
-                if (dependencyMatcher.matches(getCursor())) {
-                    if (optionalGroupId.isPresent()) {
-                        String groupId = optionalGroupId.get();
-                        String artifactId = optionalArtifactId.orElse(null);
-                        tag = maybeUpdate(acc, tag, groupId, artifactId, ctx);
+            private TreeVisitor<?, ExecutionContext> mavenVisitor() {
+                final XPathMatcher dependencyMatcher = new XPathMatcher("//dependencies/dependency");
+                return new MavenIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                        tag = super.visitTag(tag, ctx);
+                        Optional<String> optionalGroupId = tag.getChildValue("groupId");
+                        Optional<String> optionalArtifactId = tag.getChildValue("artifactId");
+                        if (dependencyMatcher.matches(getCursor())) {
+                            if (optionalGroupId.isPresent()) {
+                                String groupId = optionalGroupId.get();
+                                String artifactId = optionalArtifactId.orElse(null);
+                                tag = maybeUpdate(tag, groupId, artifactId, ctx);
+                            }
+                        } else if (isPluginTag()) {
+                            if (optionalArtifactId.isPresent()) {
+                                String groupId = tag.getChildValue("groupId").orElse("org.apache.maven.plugins");
+                                String artifactId = optionalArtifactId.get();
+                                tag = maybeUpdate(tag, groupId, artifactId, ctx);
+                            }
+                        }
+                        return tag;
                     }
-                } else if (isPluginTag()) {
-                    if (optionalArtifactId.isPresent()) {
-                        String groupId = tag.getChildValue("groupId").orElse("org.apache.maven.plugins");
-                        String artifactId = optionalArtifactId.get();
-                        tag = maybeUpdate(acc, tag, groupId, artifactId, ctx);
+
+                    private <T extends Tree> T maybeUpdate(T tree, String groupId, @Nullable String artifactId, ExecutionContext ctx) {
+                        Relocation relocation = getRelocation(groupId, artifactId);
+                        if (relocation != null) {
+                            insertRow(groupId, artifactId, relocation, ctx);
+                            if (Boolean.TRUE.equals(changeDependencies) && artifactId != null) {
+                                String newGroupId = relocation.getTo().getGroupId();
+                                String newArtifactId = Optional.ofNullable(relocation.getTo().getArtifactId()).orElse(artifactId);
+                                doAfterVisit(new ChangeDependency(
```

Comment:
This `doAfterVisit` is the reason we can't for now have a shared `maybeUpdate` between the Maven and Gradle visitors. 🤔